hazelcast/src/test/java/com/hazelcast/executor/ExecutorServiceTest.java

```diff
@@ -94,6 +95,31 @@ private IExecutorService createSingleNodeExecutorService(String name, int poolSi
         return instance.getExecutorService(name);
     }
 
+    @Test
+    public void testAndThen() throws ExecutionException, InterruptedException {
+        TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory(2);
+        HazelcastInstance instance1 = factory.newHazelcastInstance();
+        HazelcastInstance instance2 = factory.newHazelcastInstance();
+        String name = randomString();
+        IExecutorService executorService = instance2.getExecutorService(name);
+        BasicTestTask task = new BasicTestTask();
+        String key = generateKeyOwnedBy(instance1);
+        ICompletableFuture<String> future = (ICompletableFuture<String>) executorService.submitToKeyOwner(task, key);
+        final CountDownLatch latch = new CountDownLatch(1);
+        future.andThen(new ExecutionCallback<String>() {
+            @Override
+            public void onResponse(String response) {
+                latch.countDown();
```

Comment:
One minor thing, we might add an assertion here to check whether response is correct or not like below.
 
`assertEquals(BasicTestTask.RESULT, response);`